### PR TITLE
Added new settings to logcollector module, ignore and restrict

### PR DESCRIPTION
**Pull request by jotacarma90**
## Description

This PR is to add the description of the two new settings in the logcollector reference. Added description, example cases and allowed value.
Issue related: https://github.com/wazuh/wazuh/issues/5628
PR related: https://github.com/wazuh/wazuh/pull/14782

## Checks
- [x] It compiles without warnings.
- [x] Spelling and grammar. 
- [x] Used impersonal speech. 
- [x] Used uppercase only on nouns. 
- [ ] Updated the `redirect.js` script if necessary (check [this guide](https://github.com/wazuh/wazuh-documentation/blob/master/NEW_RELEASE.md)).

### CODE REVIEW

**Comment by branchnetconsulting** [comment]
I'd recommend clarifying what happens if multiple `<restrict_log>` lines or multiple `<ignore_log>` lines are used in the same `<localfile>` section.  Will the regexes for the matching tag lines just be concatenated or will the separate matching tag lines be interpreted as being in an OR relationship?  

It would also be good to state that if `<restrict_log>` and `<ignore_log>` are both used in the same `<localfile>` section, that `<ignore_log>` takes precedence when the ignore and restrict pattern both match the same line (assuming that is the behavior).

File: source/user-manual/reference/ossec-conf/localfile.rst [hunk 1]
```diff
@@ -34,6 +34,8 @@ Options
 - `exclude`_
 - `reconnect_time`_
 - `multiline_regex`_
+- `ignore_log`_
```
Comment (chemamartinez) [inline source/user-manual/reference/ossec-conf/localfile.rst:37]:
```suggestion
- `ignore`_
```
Please apply this suggestion to the whole PR.

File: source/user-manual/reference/ossec-conf/localfile.rst [hunk 2]
```diff
@@ -34,6 +34,8 @@ Options
...
+- `restrict_log`_
```
Comment (chemamartinez) [inline source/user-manual/reference/ossec-conf/localfile.rst:38]:
```suggestion
- `restrict`_
```

**Review by chemamartinez — CHANGES_REQUESTED** [review]

**Comment by jotacarma90** [comment]
Hi @branchnetconsulting 
Indeed we should clarify some points, I have updated the documentation and implemented some new features:
- Choose the type of regex expression (osregex, osmatch, pcre2. The default is pcre2).
- Possibility to use ignore and restrict multiple times (OR for ignore, AND for restrict).
- In case any ignore expression and any restrict expression are identical, no log will be processed. I have clarified it also in the docu.

Thank you very much!

File: source/user-manual/reference/ossec-conf/localfile.rst [hunk 3]
```diff
@@ -595,6 +597,96 @@ For example, we may want to read a Python Traceback output as one single log, re
       <multiline_regex replace="wspace">^Traceback</multiline_regex>
    </localfile>
 
+ignore
+^^^^^^^^^^^^^^^
+
+This allows you to configure a pcre2 expression to ignore specific log lines. Multiple expressions can be configured by repeating the ignore label. The log will be ignored as soon as one of the expressions matches, OR behavior.
```
Comment (chemamartinez) [inline source/user-manual/reference/ossec-conf/localfile.rst:603]: - Better using impersonal sentences (avoid using `This allows you`).
- pcre2 is uppercase
- Let's simplify the description
```suggestion
Specify a regular expression to ignore log lines or command outputs when matching. Whether several `ignore` labels are defined, entries are ignored when matching any of the specified ones.
```

File: source/user-manual/reference/ossec-conf/localfile.rst [hunk 4]
```diff
@@ -595,6 +597,96 @@ For example, we may want to read a Python Traceback output as one single log, re
...
+
++--------------------+---------------------------------------------------------------+
+| **Default Value**  | n/a                                                           |
++--------------------+---------------------------------------------------------------+
+| **Allowed values** | Any `regex <regex.html#regex-os-regex-syntax>`_,              |
+|                    | `sregex <regex.html#sregex-os-match-syntax>`_ or              |
+|                    | `pcre2 <regex.html#pcre2-syntax>`_ expression.                |
```
Comment (chemamartinez) [inline source/user-manual/reference/ossec-conf/localfile.rst:610]:
```suggestion
|                    | `PCRE2 <regex.html#pcre2-syntax>`_ expression.                |
```

File: source/user-manual/reference/ossec-conf/localfile.rst [hunk 5]
```diff
@@ -595,6 +597,96 @@ For example, we may want to read a Python Traceback output as one single log, re
...
++--------------------+---------------------------------------------------------------+
+
+The attribute type is optional.
```
Comment (chemamartinez) [inline source/user-manual/reference/ossec-conf/localfile.rst:613]:
```suggestion
Use the `type` attribute to define the regular expression type. By default, PCRE2 is applied.
```

File: source/user-manual/reference/ossec-conf/localfile.rst [hunk 6]
```diff
@@ -595,6 +597,96 @@ For example, we may want to read a Python Traceback output as one single log, re
...
+
+  <localfile>
+      <log_format>syslog</log_format>
+      <location>/custom/file/path</location>
+      <restrict type="osregex">username_\.+</restrict>
+      <restrict type="osregex">Jun\.+</restrict>
+  </localfile>
+
+.. note::
+  For formats that group multiple lines, the entire group will be counted as a single log to check if it should be restricted.
```
Comment (chemamartinez) [inline source/user-manual/reference/ossec-conf/localfile.rst:682]: Better use present than future.
```suggestion
  For formats that group multiple lines, the entire group is treated as a single log when evaluating the regex.
```

File: source/user-manual/reference/ossec-conf/localfile.rst [hunk 7]
```diff
@@ -595,6 +597,96 @@ For example, we may want to read a Python Traceback output as one single log, re
...
+
+.. note::
+  The ignore label is checked before the restrict label, so if you set identical expressions for both settings, no log will be processed.
```
Comment (chemamartinez) [inline source/user-manual/reference/ossec-conf/localfile.rst:685]:
```suggestion
  Whether the same log entry matches an ignore and also a restrict configured for the same `localfile`, the entry is discarded. In other words, the `ignore` has precedence over `restrict`. Said that, if the same expression is defined in both `ignore` and `restrict`, no log will be processed for that `localfile`.
```

File: source/user-manual/reference/ossec-conf/localfile.rst [hunk 8]
```diff
@@ -595,6 +597,96 @@ For example, we may want to read a Python Traceback output as one single log, re
...
+
+.. note::
+  On Windows, the ``eventchannel`` format already provides a way to restrict logs through queries, so this setting ``does not apply`` to this format.
```
Comment (chemamartinez) [inline source/user-manual/reference/ossec-conf/localfile.rst:688]:
```suggestion
  The `eventchannel` format already provides a way to filter logs through queries. Therefore, `ignore` and `restrict` settings don't apply to this format.
```

File: source/user-manual/reference/ossec-conf/localfile.rst [hunk 9]
```diff
@@ -595,6 +597,96 @@ For example, we may want to read a Python Traceback output as one single log, re
...
+.. code-block:: xml
+
+  <localfile>
+      <log_format>audit</log_format>
+      <location>/var/log/audit/audit.log</location>
+      <ignore type="osregex">type=\.+_CHANGE</ignore>
+      <ignore type="osregex">type=CONFIG_\.+</ignore>
+  </localfile>
+
+.. note::
```
Comment (chemamartinez) [inline source/user-manual/reference/ossec-conf/localfile.rst:636]: These three notes are duplicated, it is cleaner if we add it only once at the end of the `restrict` section.

File: source/user-manual/reference/ossec-conf/localfile.rst [hunk 10]
```diff
@@ -595,6 +597,96 @@ For example, we may want to read a Python Traceback output as one single log, re
...
+  The ignore label is checked before the restrict label, so if you set identical expressions for both settings, no log will be processed.
+
+.. note::
+  On Windows, the ``eventchannel`` format already provides a way to ignore logs through queries, so this setting ``does not apply`` to this format.
+
+restrict
+^^^^^^^^^^^^^^^
+
+This allows you to configure a pcre2 expression to restrict specific log lines, avoiding to generate alerts for those logs that do not match. Multiple expressions can be configured by repeating the restrict label. The log will be processed only if all expressions match.
```
Comment (chemamartinez) [inline source/user-manual/reference/ossec-conf/localfile.rst:648]:
```suggestion
Specify a regular expression to restrict processed log lines or command outputs. Whether several `restrict` labels are defined, entries are processed when matching all of them.
```

File: source/user-manual/reference/ossec-conf/localfile.rst [hunk 11]
```diff
@@ -595,6 +597,96 @@ For example, we may want to read a Python Traceback output as one single log, re
...
+
++--------------------+---------------------------------------------------------------+
+| **Default Value**  | n/a                                                           |
++--------------------+---------------------------------------------------------------+
+| **Allowed values** | Any `regex <regex.html#regex-os-regex-syntax>`_,              |
+|                    | `sregex <regex.html#sregex-os-match-syntax>`_ or              |
+|                    | `pcre2 <regex.html#pcre2-syntax>`_ expression.                |
```
Comment (chemamartinez) [inline source/user-manual/reference/ossec-conf/localfile.rst:655]:
```suggestion
|                    | `PCRE2 <regex.html#pcre2-syntax>`_ expression.                |
```

File: source/user-manual/reference/ossec-conf/localfile.rst [hunk 12]
```diff
@@ -595,6 +597,96 @@ For example, we may want to read a Python Traceback output as one single log, re
...
++--------------------+---------------------------------------------------------------+
+
+The attribute type is optional.
```
Comment (chemamartinez) [inline source/user-manual/reference/ossec-conf/localfile.rst:658]:
```suggestion
Use the `type` attribute to define the regular expression type. By default, PCRE2 is applied.
```

File: source/user-manual/reference/ossec-conf/localfile.rst [hunk 13]
```diff
@@ -595,6 +597,96 @@ For example, we may want to read a Python Traceback output as one single log, re
...
+
++-------------+-----------------------------------------+-------------+---------------+
```
Comment (chemamartinez) [inline source/user-manual/reference/ossec-conf/localfile.rst:615]: Use this table format:
image

File: source/user-manual/reference/ossec-conf/localfile.rst [hunk 14]
```diff
@@ -595,6 +597,96 @@ For example, we may want to read a Python Traceback output as one single log, re
...
+
+For example, to restrict syslog events related to a particular user name:
+
+.. code-block:: xml
+
+  <localfile>
+      <log_format>syslog</log_format>
+      <location>/custom/file/path</location>
+      <restrict type="osregex">username_\.+</restrict>
```
Comment (chemamartinez) [inline source/user-manual/reference/ossec-conf/localfile.rst:677]: Change the type of this `restrict` to show that types can be different in the same `localfile`.

File: source/user-manual/reference/ossec-conf/localfile.rst [hunk 15]
```diff
@@ -595,6 +597,96 @@ For example, we may want to read a Python Traceback output as one single log, re
...
++-------------+-----------------------------------------+-------------+---------------+
+
+For example, to ignore events related to configuration changes in the audit log:
+
+.. code-block:: xml
+
+  <localfile>
+      <log_format>audit</log_format>
+      <location>/var/log/audit/audit.log</location>
+      <ignore type="osregex">type=\.+_CHANGE</ignore>
```
Comment (chemamartinez) [inline source/user-manual/reference/ossec-conf/localfile.rst:632]: Change the type of this `ignore` to show that types can be different in the same `localfile`.

**Review by chemamartinez — CHANGES_REQUESTED** [review]

**Review by chemamartinez — APPROVED** [review]
LGTM!

**Review by Arcapas — APPROVED** [review]
LGTM

**Pull request merged**